src/auth/googleauth.ts

```diff
@@ -108,9 +123,22 @@ export class GoogleAuth {
 
   /**
    * Obtains the default project ID for the application..
-   * @param {function=} callback Optional callback.
+   * @param callback Optional callback
+   * @returns Promise that resolves with project Id (if used without callback)
    */
-  getDefaultProjectId(callback: ProjectIdCallback) {
+  getDefaultProjectId(): Promise<string>;
+  getDefaultProjectId(callback: ProjectIdCallback): void;
+  getDefaultProjectId(callback?: ProjectIdCallback): Promise<string|null>|void {
```

Comment:
nit: I'm glad that you're expanding the API here to better support promises, but in such large refactor PRs it is better to have a pure refactor CL that is easier to review; and follow-up with new functionality.

Reply:
I'm usually better about this, apologies.  I tried to bite off too much with this one.